gbif-occurrence-linkage/main.py

```diff
@@ -0,0 +1,161 @@
+import json
+import logging
+import os
+import uuid
+from datetime import datetime, timezone
+from typing import Dict
+
+import requests
+from kafka import KafkaConsumer, KafkaProducer
+
+logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
+
+
+def start_kafka() -> None:
+    """
+    Start a kafka listener and process the messages by unpacking the image.
+    When done it will republish the object, so it can be validated and storage by the processing service
+    """
+    consumer = KafkaConsumer(os.environ.get('KAFKA_CONSUMER_TOPIC'),
+                             group_id=os.environ.get('KAFKA_CONSUMER_GROUP'),
+                             bootstrap_servers=[os.environ.get('KAFKA_CONSUMER_HOST')],
+                             value_deserializer=lambda m: json.loads(m.decode('utf-8')),
+                             enable_auto_commit=True)
+    producer = KafkaProducer(bootstrap_servers=[os.environ.get('KAFKA_PRODUCER_HOST')],
+                             value_serializer=lambda m: json.dumps(m).encode('utf-8'))
+    for msg in consumer:
+        logging.info('Received message: ' + str(msg.value))
+        json_value = msg.value
+        specimen_data = json_value['digitalSpecimen']['data']
+        result = run_api_call(specimen_data)
+        if result is not None:
```

Comment:
It doesn't look like result will be None on an error, did you mean to check `result[gbif_id] `here?

Reply:
Good catch, this will also generate an annotation. Yes think I started doubting what to do when there are no matches. This is information so should it be an annotation or should this be ignored. Will ignore for now...